clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -163,6 +163,15 @@ else if (maxTimestamp == LogEntry.NO_TIMESTAMP)
             return new RecordsInfo(maxTimestamp, compressionType == CompressionType.NONE ? offsetOfMaxTimestamp : lastOffset);
     }
 
+    /**
+     * We set the sequence of the batch just before it is about to be sent out.
+     * @param baseSequence
+     */
+    public void closeWithSequence(int baseSequence) {
+        this.baseSequence = baseSequence;
```

Comment:
This just calls down to `close`, which calls `recordsBuilder.close` which does nothing  if`builtRecords`is not null. Close could be called multiple times already, so I don't think starting throw an `IllegalStateException` is a good idea.